### PR TITLE
Use unsigned char for PropertyType enum

### PR DESCRIPTION
**Pull request by kristiandupont**
..also some small fixes to make it compile on Windows

### CODE REVIEW

**Comment by fealebenpae** [comment]
👍

File: wrappers/src/schema_cs.cpp [hunk 1]
```diff
@@ -62,7 +62,7 @@ REALM_EXPORT Schema* schema_create(SchemaObject* objects, int objects_length, Sc
                 Property p;
                 p.name = property.name;
                 p.type = property.type;
-                p.object_type = property.object_type ?: "";
+                p.object_type = property.object_type ? property.object_type : "";
```
Comment (AndyDentFree) [inline wrappers/src/schema_cs.cpp]: Was this just an aesthetic change?

Comment (fealebenpae) [inline wrappers/src/schema_cs.cpp]: Apparently MSVC doesn't like the other syntax. Functionally it's equivalent, though.

Comment (tgoyne) [inline wrappers/src/schema_cs.cpp]: `?:` is unfortunately a (rather useful) GNU extension.

**Pull request merged**